### ISSUE TITLE
Bug 11: Incorrect RSI Calculation Formula

### ISSUE THREAD

**Issue by dawdameet** [dawdameet/team-finbugs-fintech#11]
## 🐛 Bug Description
The RSI (Relative Strength Index) feature is computed as NaN or nonsensical negative values. This corrupts the feature data for all ML models, leading to poor predictions.

## ✅ Expected Behavior
RSI should be a value between 0 and 100, calculated from average gains and average losses over a window period.

## ❌ Current Behavior
The loss component is calculated incorrectly, producing negative values where positive values are needed, breaking the RSI formula.

## 📁 Files Affected
`stock_price_predictor/predictor.py`

## 🎯 Difficulty: HARD
**Points: 30**

## 🔍 Reproduction Steps
```
1. Run the predictor and examine the features DataFrame
2. Notice RSI values are NaN or negative
3. Examine the create_features method
4. Look at the RSI calculation
5. Understand that RSI needs average gain and average loss
6. Check the 'loss' calculation using .where()
7. Identify that loss values are negative when they should be positive
8. Negate the loss expression
```

## 💡 Hints
Negate the expression: -delta.where(delta < 0, 0)

## 🔧 Solution (Hidden)
<details>
<summary>Click to reveal solution</summary>

```
Change 'loss = (delta.where(delta < 0, 0)).rolling(window=14).mean()' to 'loss = (-delta.where(delta < 0, 0)).rolling(window=14).mean()'
```
</details>

---
**Bug ID**: 11  
**Module**: FinBugs (Fintech AI)  
**Status**: 🔴 UNRESOLVED
